extensions/typescript-language-features/src/server.ts

```diff
@@ -83,6 +84,15 @@ class RequestQueue {
 	}
 
 	public push(item: RequestItem): void {
+		// insert before existing lowPriority requestItems in the queue.
+		if (!item.lowPriority && this.length) {
+			const lowPriorityRequests: Array<RequestItem> = this.queue.filter(item => item.lowPriority);
```

Comment:
This works but is not very efficient since it iterates over the queue multiple times and creates temporary objects. I recommend just using a loop that starts at the end of the queue and looking forward from there to find the last index of a regular priority requests. That's also not as efficient it could be but it won't be too bad for small queue sizes

Reply:
Done